status/projects.py

```diff
@@ -588,6 +588,64 @@ def get_caliper_image(url):
             print(message)
             raise tornado.web.HTTPError(404, reason='Error fetching caliper images')
 
+
+class ImagesDownloadHandler(SafeHandler):
+    """serves caliper/fragment_analyzer images in a zip archive.
+    Called through /api/v1/download_images/PROJECT/[image_type]
+    where image types are currently frag_an, caliper_libval or caliper_initial_qc
+    """
+
+    def post(self, project, type):
+        from io import BytesIO
+        import zipfile as zp
+
+        name = ''
+        if type == 'frag_an':
+            view = 'project/frag_an_links'
+            name = 'InitialQCFragmentAnalyser'
+        else:
+            view = 'project/caliper_links'
+            if 'libval' in type:
+                name = 'LibraryValidationCaliper'
+            elif 'initial_qc' in type:
+                name = 'InitialQCCaliper'
+
+        sample_view = self.application.projects_db.view(view)
+        result = sample_view[project]
+        try:
+            data = result.rows[0].value
+        except TypeError:
+            #can be triggered by the data.get().get() calls.
+            raise tornado.web.HTTPError(404, reason='No caliper image found')
+
+        fileName = '{}_{}_images.zip'.format(project, name)
+        f = BytesIO()
+        num_files = 0
+        with zp.ZipFile(f, "w") as zf:
```

Comment:
I think you could do `with zp.ZipFile(f, "w") as zf, BytesIO() as f:` should save you a bit of precious lines of code, but it's also more certain that the close is always called right?

Reply:
I had some issues with corrupted zip files when I tried it. We can put it on the list of improvements?